server/lib/report_server_web/live/report_run_live/show.ex

```diff
@@ -97,18 +109,22 @@ defmodule ReportServerWeb.ReportRunLive.Show do
     end
   end
 
-  defp run_report(nil, report_run, _sort_columns, _row_limit, _user) do
+  defp run_report(nil, report_run, _sort_columns, _row_limit, _live_view_pid) do
     error = "Unable to find report: #{report_run.report_slug}"
     Logger.error(error)
     {:error, error}
   end
 
-  defp run_report(report = %Report{}, report_run = %ReportRun{}, sort_columns, row_limit, user = %User{}) do
-    with {:ok, query} <- report.get_query.(report_run.report_filter),
-      ordered_query <- ReportQuery.add_sort_columns(query, sort_columns),
-      sql <- ReportQuery.get_sql(ordered_query, row_limit),
-      {:ok, results} <- PortalDbs.query(user.portal_server, sql, []) do
-        {:ok, %{report_results: results}}
+  defp run_report(report = %Report{type: :athena}, report_run = %ReportRun{id: report_run_id, athena_query_id: nil}, _sort_columns, _row_limit, live_view_pid) do
+    with {:ok, query} <- report.get_query.(report_run.report_filter, report_run.user),
+         {:ok, sql} <- ReportQuery.get_sql(query),
+         {:ok, athena_query_id, athena_query_state} <- AthenaDB.query(sql, report_run_id, report_run.user),
+         {:ok, _report_run} <- Reports.update_report_run(report_run, %{athena_query_id: athena_query_id, athena_query_state: athena_query_state}) do
+
+      send(live_view_pid, :poll_query_state)
+
+      # return nil - the template will use the report_run to provide status updates
+      {:ok, %{report_results: nil}}
 
     else
       {:error, error} ->
```

Comment:
As noted in Slack, athena seems to return `:exit` rather than `:error` for at least some kinds of problems.